analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Sonar/LocksShouldBeReleased.Mutex.cs

```diff
@@ -0,0 +1,261 @@
+﻿using System;
+using System.Threading;
+
+// https://docs.microsoft.com/en-us/dotnet/standard/threading/mutexes
+internal class Foo
+{
+    bool cond;
+    public Mutex instanceMutex;
+    public static Mutex staticMutex;
+
+    // Note that dispose does not release the resource
+    public void Noncompliant(Mutex paramMutex, Mutex paramMutex2, Foo foo)
+    {
+        var m0 = new Mutex(true, "bar", out var m0WasCreated); // FN
+
+        var m1 = new Mutex(false);
+        m1.WaitOne(); // FN
+
+        var m2 = new Mutex(false, "qix", out var m2WasCreated);
+        m2.WaitOne(); // FN
+
+        var m3 = Mutex.OpenExisting("x");
+        m3.WaitOne(); // FN
+
+        foo.instanceMutex.WaitOne(); // FN
+
+        Foo.staticMutex.WaitOne(); // FN
+
+        if (cond)
+        {
+            m0.ReleaseMutex();
+            m1.ReleaseMutex();
+            m2.ReleaseMutex();
+            m3.ReleaseMutex();
+            foo.instanceMutex.ReleaseMutex();
+            Foo.staticMutex.ReleaseMutex();
+        }
+        m0.Dispose();
+        m1.Dispose();
+        m2.Dispose();
+        m3.Dispose();
+
+        using (var mutexInUsing = new Mutex(true, "foo")) // FN
+        {
+            // 'true' means it owns the mutex if no exception gets thrown
```

Comment:
I don't understand this comment as it looks like related to `cond` variable. Should it be on line 43, referring to the first argument?